asserts/asserts.go

```diff
@@ -278,6 +278,62 @@ func (ref *Ref) Resolve(find func(assertType *AssertionType, headers map[string]
 
 const RevisionNotKnown = -1
 
+// AtSequence represents a sequence forming assertion at a given sequence possibly
+// <= 0 (meaning not specified) and revision, possibly not known (RevisionNotKnown).
+type AtSequence struct {
+	Type        *AssertionType
+	SequenceKey []string
+	Sequence    int
+	Pinned      bool
+	Revision    int
+}
+
+// Unique returns a unique string representing the sequence by its sequence key
+// that can be used as a key in maps.
+func (at *AtSequence) Unique() string {
+	return fmt.Sprintf("%s/%s", at.Type.Name, strings.Join(at.SequenceKey, "/"))
+}
+
+func (at *AtSequence) String() string {
```

Comment:
Ref are used in many more contexts than these, I think for sequences and AtSequence we can probably use a simplified string representation:

validation-set account-id/name(/|=)8  at revision #

basically we omit "series" if that's the first key in primary key,  and we separate things with /  except sequence we separate it with = if Pinned is true

Reply:
Done